src/azul/indexer/aggregate.py

```diff
@@ -163,6 +167,30 @@ def get(self):
         return thaw(super().get())
 
 
+class DictAccumulator(Accumulator):
+    """
+    An accumulator of dictionaries. The key should return a unique value the
+    accumulation of dictionaries can be sorted by. (eg. `itemgetter('uuid')`)
+    """
+
+    def __init__(self, key: Callable):
+        self.key = key
+        self.value = {}
+
+    def accumulate(self, value):
+        require(isinstance(value, dict), value)
```

Comment:
Other than this, nothing else requires `value` to be a dict. It is never used as a dict. It could be anything. I still like the name DictAccumulator, not because the accumulated values are dicts but because their are accumulated in a dict. In a way this is a generalized SetAccumulator. DictAccumulator can replace a SetAccumulator by using the identity function for the key (`lambda _: _`). Not saying we should do that, but it would help if the documentation mentioned that.